### PR TITLE
Add support for JSON extractor

### PR DESCRIPTION
**Pull request by joschi**
Refs Graylog2/graylog2-server#632, Graylog2/graylog2-server#1355

### CODE REVIEW

File: app/views/system/inputs/extractors/configurations/json.scala.html [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    if(extractor == null) {
+        "="
+    } else {
+        String.valueOf(extractor.getExtractorConfig.get("kv_separator"))
+    }
+}
+
+<input type="hidden" name="extractor_type" value="json" />
+<div class="form-group">
+    <label for="flatten">Flatten structures</label>
```
Comment (edmundoa) [inline app/views/system/inputs/extractors/configurations/json.scala.html]: This checkbox looks weird, I would use something like this instead:
```
<div class="row">
    <div class="col-md-12">
        <div class="checkbox">
            <label>
                <input name="flatten" id="flatten" type="checkbox" value="flatten" @if(flatten) { checked="checked" } />
                Flatten structures
            </label>
        </div>
        <span class="help-block">
            Whether to flatten JSON objects into a single message field or to expand into multiple fields.
        </span>
    </div>
</div>
```

File: app/views/system/inputs/extractors/configurations/json.scala.html [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        <div class="col-md-5">
+            <input name="kv_separator" id="kv_separator" type="text" class="form-control" value="@kv_separator" required />
+        </div>
+    </div>
+    <span class="help-block">
+        What string to use when concatenating key/value pairs of a JSON object (only used if flattened).
+    </span>
+</div>
+
+<button class="btn btn-info xtrc-try-json" type="button">Try!</button>
```
Comment (edmundoa) [inline app/views/system/inputs/extractors/configurations/json.scala.html]: The try button doesn't work, I think there's some javascript missing. I think you will be happy to look into `extractors.js`. Or maybe not.

**Comment by edmundoa** [comment]
Other than the inline issues, we are also missing a link to the new extractor on the search result page, specifically in `javascript/src/components/search/MessageFieldDescription.jsx`.

File: app/views/system/inputs/extractors/configurations/json.scala.html (hunk 2 above)
Comment (joschi) [inline app/views/system/inputs/extractors/configurations/json.scala.html]: computering

**Comment by joschi** [comment]
> Other than the inline issues, we are also missing a link to the new extractor on the search result page, specifically in `javascript/src/components/search/MessageFieldDescription.jsx`.

I didn't even know we had this feature…
¯\_(ツ)_/¯

**Comment by edmundoa** [comment]
Looking good now! :shipit:

**Pull request merged**